src/platform/Linux/MdnsImpl.cpp

```diff
@@ -540,25 +541,45 @@ CHIP_ERROR MdnsAvahi::Browse(const char * type, MdnsServiceProtocol protocol, ch
     return browser == nullptr ? CHIP_ERROR_INTERNAL : CHIP_NO_ERROR;
 }
 
-MdnsServiceProtocol TruncateProtocolInType(char * type)
+MdnsServiceProtocol GetProtocolInType(const char * type)
 {
-    char * deliminator           = strrchr(type, '.');
-    MdnsServiceProtocol protocol = MdnsServiceProtocol::kMdnsProtocolUnknown;
+    const char * deliminator = strrchr(type, '.');
 
-    if (deliminator != NULL)
+    if (deliminator == NULL)
     {
-        if (strcmp("._tcp", deliminator) == 0)
-        {
-            protocol     = MdnsServiceProtocol::kMdnsProtocolTcp;
-            *deliminator = 0;
-        }
-        else if (strcmp("._udp", deliminator) == 0)
-        {
-            protocol     = MdnsServiceProtocol::kMdnsProtocolUdp;
-            *deliminator = 0;
-        }
+        ChipLogError(Discovery, "Failed to find protocol in type: %s", type);
+        return MdnsServiceProtocol::kMdnsProtocolUnknown;
+    }
+
+    if (strcmp("._tcp", deliminator) == 0)
+    {
+        return MdnsServiceProtocol::kMdnsProtocolTcp;
+    }
+    if (strcmp("._udp", deliminator) == 0)
+    {
+        return MdnsServiceProtocol::kMdnsProtocolUdp;
+    }
+
+    ChipLogError(Discovery, "Unknown protocol in type: %s", type);
+    return MdnsServiceProtocol::kMdnsProtocolUnknown;
+}
+
+/// Copies the type from a string containing both type and protocol
+///
+/// e.g. if input is "foo.bar", output is "foo", input is 'a.b._tcp", oputput is "a.b"
```

Comment:
o~p~utput (Although ‘oputput’ is not bad; I could also accept replacing ‘input’ with ‘igetget’.)